packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -376,6 +376,13 @@ export function createVM<HostNode, HostElement>(
     vm.shadowMode = computeShadowMode(def, vm.owner, renderer);
     vm.tro = getTemplateReactiveObserver(vm);
 
+    if (isReportingEnabled()) {
+        report(ReportingEventId.ShadowModeUsage, {
+            tagName: vm.tagName,
+            mode: vm.shadowMode,
+        });
+    }
+
```

Comment:
I checked, and this gets tree-shaken if the customer is not using the reporting API! :tada: